### PR TITLE
Use the new serializer and deserializer in the interface and adjust the C# code

### PR DESCRIPTION
**Pull request by pleroy**

### CODE REVIEW

File: base/push_deserializer.hpp [hunk 1]
```diff
@@ -65,7 +65,9 @@ class PushDeserializer {
   // Starts the deserializer, which will proceed to deserialize data into
   // |message|.  This method must be called at most once for each deserializer
   // object.
-  void Start(not_null<google::protobuf::Message*> const message);
+  //TODO(phl):comment
```
Comment (eggrobin) [inline base/push_deserializer.hpp]: TODO

File: ksp_plugin/interface.hpp [hunk 2]
```diff
@@ -276,10 +278,14 @@ XYZ CDECL principia__BubbleVelocityCorrection(Plugin const* const plugin,
 extern "C" DLLEXPORT
 double CDECL principia__current_time(Plugin const* const plugin);
 
-// |plugin| must not be null.  The caller takes ownership of the result.  No
-// transfer of ownership of |*plugin|.
+// |plugin| must not be null.  The caller takes ownership of the result, except
+// when it is null (at the end of the stream).  No transfer of ownership of
+// |*plugin|.  |*serializer| must be null on the first call and must be passed
+// unchanged to the successive calls.
```
Comment (eggrobin) [inline ksp_plugin/interface.hpp]: Detail ownership of serializer.

File: ksp_plugin_adapter/ksp_plugin_adapter.cs [hunk 3]
```diff
@@ -268,9 +268,15 @@ where is_in_physics_bubble(vessel)
     base.OnSave(node);
     if (PluginRunning()) {
       IntPtr serialization = IntPtr.Zero;
+      IntPtr serializer = IntPtr.Zero;
       try {
-        serialization = SerializePlugin(plugin_);
-        node.AddValue(kPrincipiaKey, Marshal.PtrToStringAnsi(serialization));
+        for (;;) {
+          serialization = SerializePlugin(plugin_, ref serializer);
```
Comment (eggrobin) [inline ksp_plugin_adapter/ksp_plugin_adapter.cs]: The previous serialization hasn't been deleted, you're leaking memory like a broken snowden.

File: ksp_plugin_adapter/ksp_plugin_adapter.cs [hunk 4]
```diff
@@ -282,9 +288,22 @@ where is_in_physics_bubble(vessel)
     if (node.HasValue(kPrincipiaKey)) {
       Cleanup();
       SetRotatingFrameThresholds();
-      String serialization = node.GetValue(kPrincipiaKey);
-      Log.Info("serialization is " + serialization.Length + " characters long");
-      plugin_ = DeserializePlugin(serialization, serialization.Length);
+
+      IntPtr deserializer = IntPtr.Zero;
+      plugin_ = IntPtr.Zero;
```
Comment (eggrobin) [inline ksp_plugin_adapter/ksp_plugin_adapter.cs]: Either it is already zero, or it hasn't been deleted and you're leaking memory.

File: ksp_plugin/interface.hpp (hunk 2 above)
Comment (pleroy) [inline ksp_plugin/interface.hpp]: Done.

File: ksp_plugin_adapter/ksp_plugin_adapter.cs (hunk 3 above)
Comment (pleroy) [inline ksp_plugin_adapter/ksp_plugin_adapter.cs]: Done.

File: ksp_plugin_adapter/ksp_plugin_adapter.cs (hunk 4 above)
Comment (pleroy) [inline ksp_plugin_adapter/ksp_plugin_adapter.cs]: It is already zero because of Cleanup().

File: ksp_plugin/interface.cpp [hunk 5]
```diff
@@ -32,6 +38,9 @@ namespace ksp_plugin {
 
 namespace {
 
+const int kChunkSize = 64 << 10;
+const int kNumberOfChunks = 8;
```
Comment (eggrobin) [inline ksp_plugin/interface.cpp]: `int const`

File: ksp_plugin/interface.hpp [hunk 6]
```diff
@@ -288,11 +294,18 @@ extern "C" DLLEXPORT
 void CDECL principia__DeletePluginSerialization(
     char const** const serialization);
 
-// The caller takes ownership of the result.  No transfer of ownership of
-// |*serialization|.
-extern "C" DLLEXPORT
-Plugin* CDECL principia__DeserializePlugin(char const* const serialization,
-                                           int const serialization_size);
+// The caller takes ownership of |*plugin| when it is not null.  No transfer of
```
Comment (eggrobin) [inline ksp_plugin/interface.hpp]: **plugin

**Pull request merged**